test/hotspot/jtreg/compiler/lib/ir_framework/driver/irmatching/irrule/checkattribute/parsing/RawIRNode.java

```diff
@@ -51,14 +55,85 @@ public CompilePhase defaultCompilePhase() {
+                IRNode.parseVectorNodeSize(size, type, vmInfo);
+            }
+        }
+
+        // Set default values in some cases:
+        if (!userPostfix.isValid() || !vmInfo.canTrustVectorSize()) {
+            switch (bound) {
+                case Comparison.Bound.LOWER -> {
+                    // For lower bound we check for the maximal size by default. But if we cannot trust the
+                    // vector size we at least check there are vectors of any size.
+                    if (vmInfo.canTrustVectorSize()) {
+                        // No size specified, so assume maximal size
+                        size = IRNode.VECTOR_SIZE_TAG_MAX;
+                    } else {
+                        System.out.println("WARNING: you are on a system with \"canTrustVectorSize == false\" (default Cascade Lake).");
+                        System.out.println("         The lower bound rule for \"" + node + "\" is now performed with");
+                        System.out.println("         \"IRNode.VECTOR_SIZE_TAG_ANY\" instead of \"IRNode.VECTOR_SIZE_TAG_MAX\".");
+                        size = IRNode.VECTOR_SIZE_TAG_ANY;
+                    }
+                }
+                case Comparison.Bound.UPPER -> {
+                    if (userPostfix.isValid()) {
+                        TestFormat.checkNoReport(!vmInfo.canTrustVectorSize(), "sanity");
+                        // If we have a size specified but cannot trust the size, and must check an upper
+                        // bound, this can be impossible to count correctly - if we have an incorrect size
+                        // we may count either too many nodes. We just create an impossible regex which will
+                        // always have count zero and make the upper bound rule pass.
+                        System.out.println("WARNING: you are on a system with \"canTrustVectorSize == false\" (default Cascade Lake).");
+                        System.out.println("         The upper bound rule for \"" + node + "\" cannot be checked.");
+                        throw new SuccessOnlyConstraintException("upper bound with specified size");
+                    } else {
+                        // For upper bound we check for vectors of any size by default.
+                        size = IRNode.VECTOR_SIZE_TAG_ANY;
+                    }
+                }
+                case Comparison.Bound.EQUAL -> {
+                    if (vmInfo.canTrustVectorSize()) {
+                        // No size specified, so assume maximal size
+                        size = IRNode.VECTOR_SIZE_TAG_MAX;
+                    } else {
+                        // Equal comparison to a strictly positive number would lead us to an impossible
+                        // situation: we might have to know the exact vector size or else we count too many
+                        // or too few cases. Because of this we forbid this case in general.
```

Comment:
The comment is outdated and could, for example, be updated to:
```suggestion
                        // or too few cases. We therefore skip such a constraint and treat it as success.
```